common/txmgr/broadcaster.go

```diff
@@ -493,13 +493,13 @@ func (eb *Broadcaster[CHAIN_ID, HEAD, ADDR, TX_HASH, BLOCK_HASH, SEQ, FEE]) hand
 		errType, err = eb.validateOnChainSequence(ctx, lgr, errType, err, etx, retryCount)
 	}
 
-	if errType != client.Fatal {
+	if errType != client.Fatal && errType != client.TerminallyStuck {
```

Comment:
The code here is a bit a confusing. Why not structure it like:

```
if (errType == client.Fatal || errType == client.TerminallyStuck) {
                 eb.SvcErrBuffer.Append(err)
		etx.Error = null.StringFrom(err.Error())
		return eb.saveFatallyErroredTransaction(lgr, &etx), true
}

etx.InitialBroadcastAt = &initialBroadcastAt
etx.BroadcastAt = &initialBroadcastAt
		
switch errType {
		.....
}
```

Reply:
I think the condition would always evaluate as true then. We only want to set these fields if its neither fatal nor terminally stuck.

Reply:
We could structure the code that way but I think we'd still have to include a case for `Fatal` and `TerminallyStuck` in the switch to avoid it from falling through to the default case. If we have those cases anyways, I think it makes sense to keep that logic in the switch statement

Reply:
umm if we return inside if, then it doesn't have to enter swtich and fall to default ?

Reply:
Oh true, missed that part. I can structure the code this way then. Returning early would fit the go standard.